.github/actions/build-singleapp-if-changed/action.yml

```diff
@@ -0,0 +1,37 @@
+name: Build Single App If Changed
+description: Checks if parent/pom.xml changed and builds single application if needed
+outputs:
+  parent_changed:
+    description: 'Whether parent/pom.xml changed'
+runs:
+  using: "composite"
+  steps:
+    - name: Check if parent/pom.xml changed
+      id: parent_check
+      shell: bash
+      run: |
+        echo "Base branch: main"
+        echo "PR branch: ${{ github.head_ref }}"
+        git fetch origin main
+        PARENT_CHANGED=$(git diff --name-only origin/main | grep 'parent/pom.xml' || true)
+        if [[ -n "$PARENT_CHANGED" ]]; then
+          echo "parent/pom.xml changed. Building single application."
+          echo "parent_changed=true" >> "$GITHUB_OUTPUT"
+        else
+          echo "parent/pom.xml did not change."
+          echo "parent_changed=false" >> "$GITHUB_OUTPUT"
+        fi
```

Comment:
We can leverage the existing file change detection logic for that, no need to manually reinvent this with hard-coded base branch etc: https://github.com/camunda/camunda/blob/main/.github/actions/paths-filter/action.yml (declare a new output + filter expression for `parent/pom.xml`)